osu.Game/Graphics/Carousel/Carousel.cs

```diff
@@ -738,13 +738,18 @@ private void updateDisplayedRange(DisplayRange range)
                 Scroll.SetLayoutHeight(0);
         }
 
-        private static void expirePanelImmediately(Drawable panel)
+        private void expirePanel(Drawable panel)
         {
-            panel.FinishTransforms();
-            panel.Expire();
-
             var carouselPanel = (ICarouselPanel)panel;
 
+            // expired panels should have a depth behind all other panels to make the transition not look weird.
+            Scroll.Panels.ChangeChildDepth(panel, panel.Depth + 1);
```

Comment:
This is weirdly magic... Why `+ 1` specifically? Is that enough? Looking at
https://github.com/ppy/osu/blob/45d248cd26ee4d9ab000249b95f4b8fd55dfd161/osu.Game/Screens/SelectV2/BeatmapCarouselFilterGrouping.cs#L77
https://github.com/ppy/osu/blob/45d248cd26ee4d9ab000249b95f4b8fd55dfd161/osu.Game/Screens/SelectV2/BeatmapCarouselFilterGrouping.cs#L95
https://github.com/ppy/osu/blob/45d248cd26ee4d9ab000249b95f4b8fd55dfd161/osu.Game/Graphics/Carousel/Carousel.cs#L626-L627

I'm not super sure how this works?

Reply:
Hmm, I did miss the `DepthLayer` part. Originally I had 1024 as an arbitrary big number, would you be okay with that?

Reply:
can't use that because we're adding to a non-zero value 😄